lib/sim.c

```diff
@@ -892,14 +904,18 @@ sim_update_parental_ancestry(sim_t *self, unsigned int parent,
  * Generates parental individuals and inserts them into the parent
  * buffer based on the individuals in the child_buffer. 
  */
+
 static int 
-sim_generate_parents(sim_t *self)
+sim_generate_parents(sim_t *self, int arg_flag)
```

Comment:
I think it would be better to write a new function, ``sim_generate_arg_parents``, and leave ``sim_geneate_parents`` as it is (we might even rename it ``sim_generate_spatial_parents``). We're making both parts much more complicated by combining them I think. 

In fact, I think it would be easier if we made two new functions that just did the required Kingman stuff: 
``sim_arg_coancestry_event()`` and ``sim_arg_recombination_event``. We can still reuse the basic bits like ``sim_remove_individual``, ``sim_add_individual`` and ``sim_update_parental_ancestry`` which should make it relatively straightforward.